sql/core/src/main/scala/org/apache/spark/sql/execution/window/WindowFunctionFrame.scala

```diff
@@ -123,10 +123,7 @@ abstract class OffsetWindowFunctionFrameBase(
   protected val fillDefaultValue = {
     // Collect the expressions and bind them.
     val boundExpressions = Seq.fill(ordinal)(NoOp) ++ expressions.toImmutableArraySeq.map { e =>
-      if (e.default == null) {
```

Comment:
It's not clear to me why this is not possible to happen, and I don't think it's worth the time to figure it out. This null check doesn't hurt at all.

Reply:
Got it. Maybe there is a chance in future. Let's holding this PR.